pkg/platform/runtime/runtime.go

```diff
@@ -356,3 +358,29 @@ func (r *Runtime) ExecutableDirs() (envdef.ExecutablePaths, error) {
 func IsRuntimeDir(dir string) bool {
 	return store.New(dir).HasMarker()
 }
+
+func (r *Runtime) ResolvedArtifacts() ([]artifact.Artifact, error) {
+	if r.resolvedArtifacts == nil {
+		runtimeStore := r.store
+		if runtimeStore == nil {
+			runtimeStore = store.New(r.target.Dir())
+		}
+
+		plan, err := runtimeStore.BuildPlan()
+		if err != nil {
+			return nil, errs.Wrap(err, "Unable to fetch build plan")
+		}
```

Comment:
This would still mean the buildplan gets parsed twice if it was used for other purposes (eg. runtime was updated). Given that this is a costly operation we may want to introduce some caching here too? Suggest checking with Mike, he may have some ideas/opinions.

This is based on my assumption that parsing buildplans still has a performance impact that is noticeable by users. If I'm wrong on that I'll drop this concern.

Reply:
After a discussion on Slack, Mike said "As far as calling this method goes I don’t think it’s a big performance issue. This is returning the build structure which is created by parsing the build plan, so we aren’t parsing again to get it."

The function is just unmarshaling json into a Go struct representing an already parsed build plan.